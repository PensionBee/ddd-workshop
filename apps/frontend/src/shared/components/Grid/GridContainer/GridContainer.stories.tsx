import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { Box } from "../../Box/Box";
import type { GridContainerProps } from "./GridContainer";
import { GridContainer } from "./GridContainer";

export default {
  title: "Shared/Grid/Container",
  component: GridContainer,
  parameters: {
    viewMode: "story",
    previewTabs: {
      "storybook/docs/panel": { hidden: true },
    },
  },
} as ComponentMeta<typeof GridContainer>;

const GridContainerTemplate: ComponentStory<typeof GridContainer> = (
  args: GridContainerProps
) => {
  return (
    <Box position="relative">
      <GridContainer {...args}>
        <Box backgroundColor="grey4" w="100%" h={200} />
      </GridContainer>
    </Box>
  );
};

export const GridContainerDefault = GridContainerTemplate.bind({});
GridContainerDefault.args = {
  backgroundColor: "grey8",
};
