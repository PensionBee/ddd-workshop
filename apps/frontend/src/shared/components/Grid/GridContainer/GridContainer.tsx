import { Box } from "../../Box/Box";

export type GridContainerProps = {
  /** React children */
  children: React.ReactNode;
  /** Background color */
  backgroundColor?: React.ComponentProps<typeof Box>["backgroundColor"];
  /** Border top in px */
  borderTop?: React.ComponentProps<typeof Box>["borderTop"];
  /** Border bottom in px */
  borderBottom?: React.ComponentProps<typeof Box>["borderBottom"];
  /** Border colour */
  borderColor?: React.ComponentProps<typeof Box>["borderColor"];
  /** Padding top in px */
  pt?: React.ComponentProps<typeof Box>["pt"];
  /** Padding bottom in px */
  pb?: React.ComponentProps<typeof Box>["pb"];
  /** Z Index */
  z?: React.ComponentProps<typeof Box>["z"];
};

export const GridContainer: React.FC<GridContainerProps> = ({
  children,
  backgroundColor,
  borderTop,
  borderBottom,
  borderColor,
  pt,
  pb,
  z,
  ...props
}) => {
  return (
    <Box
      data-rc="GridContainer"
      as="section"
      backgroundColor={backgroundColor}
      borderTop={borderTop}
      borderBottom={borderBottom}
      borderColor={borderColor}
      pt={pt}
      pb={pb}
      z={z}
      {...props}
    >
      <Box wMax={{ xl: 1200 }} mx={{ xs: 20, md: 28, lg: 36, xl: "auto" }}>
        {children}
      </Box>
    </Box>
  );
};
