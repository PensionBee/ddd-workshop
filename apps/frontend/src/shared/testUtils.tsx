import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
