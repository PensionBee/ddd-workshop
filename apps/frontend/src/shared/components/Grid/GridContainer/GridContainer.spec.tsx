import { render } from "@shared/testUtils";
import { vi } from "vitest";
import { GridContainer } from "./GridContainer";

describe("Container", () => {
  const containerContent = "Container Content";
  describe("Default", () => {
    it("should render correctly", () => {
      const { getByText, asFragment } = render(
        <GridContainer>{containerContent}</GridContainer>
      );

      expect(getByText(containerContent)).toBeVisible();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
