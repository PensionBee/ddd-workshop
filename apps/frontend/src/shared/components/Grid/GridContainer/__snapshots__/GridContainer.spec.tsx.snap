// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > Default > should render correctly 1`] = `
<DocumentFragment>
  <section
    class="sc-bczSft hVNnBN"
    data-rc="GridContainer"
  >
    <div
      class="sc-bczSft hlbvid"
    >
      Container Content
    </div>
  </section>
</DocumentFragment>
`;

exports[`Container Default should render correctly 1`] = `
<DocumentFragment>
  <section
    class="sc-bcXHqe bLlgzy"
    data-rc="GridContainer"
  >
    <div
      class="sc-bcXHqe kVKkQK"
    >
      Container Content
    </div>
  </section>
</DocumentFragment>
`;
